engine/src/test/java/com/datatorrent/stram/engine/StreamingContainerTest.java

```diff
@@ -94,4 +121,33 @@ public void emitTuples()
     private static final Logger logger = LoggerFactory.getLogger(CommitAwareOperator.class);
   }
 
+  private static class CommitAwareInputOperator extends CommitAwareOperator
+  {
+    public final transient DefaultOutputPort<String> output = new DefaultOutputPort<String>();
+  }
+
+  private static class CommitAwareProcessOperator extends CommitAwareOperator
+  {
+    private static final Logger logger = LoggerFactory.getLogger(CommitAwareOperator.class);
+    public static ArrayList<Long> committedWindowIds;
+    public static final synchronized List<Long> getCommittedWindowIdsContainer()
```

Comment:
OK, I see. Please consider using single CommitAwareOperator and use "name" to check whether commit was called or not. Something like:
```
public class StreamingContainerTest
{
  private static final Logger logger = LoggerFactory.getLogger(StreamingContainerTest.class);

  private static Set<String> committedWindowIds = Collections.synchronizedSet(new HashSet<String>());
  private static Set<String> checkpointedWindowIds = Collections.synchronizedSet(new HashSet<String>());

  @Test
  public void testCommitted() throws IOException, ClassNotFoundException
  {
    LogicalPlan lp = new LogicalPlan();
    String workingDir = new File("target/testCommitted").getAbsolutePath();
    lp.setAttribute(Context.OperatorContext.STORAGE_AGENT, new AsyncFSStorageAgent(workingDir, null));
    lp.setAttribute(DAGContext.CHECKPOINT_WINDOW_COUNT, 1);
    CommitAwareOperator operator = lp.addOperator("CommitAwareOperator", new CommitAwareOperator());

    StramLocalCluster lc = new StramLocalCluster(lp);
    lc.run(5000);

    /* this is not foolproof but some insurance is better than nothing */
    Assert.assertTrue("No Committed Windows", committedWindowIds.contains("CommitAwareOperator"));
  }

  private static class CommitAwareOperator extends BaseOperator implements CheckpointListener, InputOperator
  {
    private String name;
    public final transient OutputPort<Object> output = new DefaultOutputPort<>();
    public final transient InputPort<Object> input = new DefaultInputPort<Object>() {
      @Override
      public void process(Object tuple)
      {
      }
    };

    @Override
    public void setup(Context.OperatorContext context)
    {
      name = context.getName();
      super.setup(context);
    }

    @Override
    public void checkpointed(long windowId)
    {
      checkpointedWindowIds.add(name);
      logger.info("{} checkpointed {}", name, windowId);
    }

    @Override
    public void committed(long windowId)
    {
      committedWindowIds.add(name);
      logger.info("{} committed {}", name, windowId);
    }

    @Override
    public void emitTuples()
    {
    }
  }
}
```